### DISCUSSION TITLE
Context menu contribution

### DISCUSSION THREAD

**Discussion by anpin** [CommunityToolkit/Maui#27] [Q&A] [answered]
Hi! I haven't really tried MAUI yet, but I was wondering if it makes sense to port my existing [context menu plugin for XF](https://github.com/anpin/ContextMenuContainer) to MAUI as a part of this community toolkit or if something would be implemented natively in the MAUI itself and porting it would be just a waste of time?

**Comment by mhrastegari** [comment]
This would be really AWESOME!!!

**Comment by pictos** [comment]
Hey, @anpin thanks for your interest in our Maui Community Toolkit. Can you tell how your plugin is different from our PopUp control? 

I believe that we can archive the same goal with the PopUp + anchor feature. Maybe your plugin has more properties that make the creation of the context menu easier and we can expand our PopUp control with those properties. What do you think?

**Comment by anpin** [comment]
Hi @pictos that is what I’m asking essentially, if there is something already implemented to cover such functionality in the toolkit. I will take a look at the popup control thanks.

**Comment by anpin** [comment]
I briefly looked over this repo and main dotnet/MAUI repo and I couldn't find the PopUp control.  @pictos can you point me in the right direction please?

**Comment by IeuanWalker** [comment]
@anpin doens look like it's been ported over yet - https://github.com/xamarin/XamarinCommunityToolkit/tree/main/src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup

**Comment by GeorgeLeithead** [comment]
I believe that Popup has now (for a while actually) been ported.  [Samples source](https://github.com/CommunityToolkit/Maui/tree/main/samples/CommunityToolkit.Maui.Sample/Pages/Views/Popup).

**Comment by VladislavAntonyuk** [comment, marked as answer]
Context Menu is a part of .NET MAUI 7, so it won't be a part of the CommunityToolkit.

**Comment by CliffAgius** [comment]
Closed as answered
